---
title: "Seurat-Object-v0.0.1"
author: "SawchukS"
date: "2025-06-13"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      eval = FALSE,
                      out.width = "100%",
                      warning = FALSE,
                      cache = TRUE)

library(metap)
library(feather)
library(Seurat)
library(ggplot2)

mtgFolder = "~/proj/proj5HT/den/GreatApes_Macaque_NCBI/"
patchseqFolder = "~/proj/proj5HT/den/macaque/MTG/GreatApes_Macaque_NCBI_RSC-204-386_map_full/"

HTRgenesINT = c(
  "HTR1A",
  "HTR1B",
  "HTR1D",
  "HTR1E",
  "HTR1F",
  "HTR2A",
  "HTR2B",
  "HTR2C",
  "HTR3A",
  "HTR3B",
  "HTR3C",
  "HTR3D",
  "HTR3E",
  "HTR4",
  "HTR5A",
  "HTR6",
  "HTR7"
)

HTRgenes12 = c(
  "HTR1A",
  "HTR1B",
  "HTR1D",
  "HTR1E",
  "HTR1F",
  "HTR2A",
  "HTR2B",
  "HTR2C"
)
```


```{r eval = TRUE}
brain = readRDS("~/proj/proj5HT/data-raw/Seurat-brain.rds")
```


# Loading the data

```{r fromFeather}
subclass<-c('L5 IT', 'L5 ET', "L2/3 IT")

#### Feather for 10x data ####
aMTG <- read_feather(paste(mtgFolder, "anno.feather",sep=""))
dMTG<-  read_feather(paste(mtgFolder, "data.feather",sep=""))

# Match by ID
rownames(dMTG) = dMTG$gene
dMTG$gene = NULL

#aMTG<- aMTG[match(names(dMTG),aMTG$sample_id),]

# Select Subclass
dMTG<-dMTG[,is.element(aMTG$subclass_label,subclass)]
aMTG<-aMTG[is.element(aMTG$subclass_label,subclass),]



#### Feather for patch data ####
aPatch<-read_feather(paste(patchseqFolder, "anno.feather",sep=""))
dPatch<- read_feather(paste(patchseqFolder, "data.feather", sep =""))
# Match by ID
aPatch<- aPatch[match(dPatch$sample_id,aPatch$sample_id),]
# Select Subclass
dPatch = dPatch[is.element(aPatch$subclass_Tree_label, subclass),]
aPatch = aPatch[is.element(aPatch$subclass_Tree_label, subclass),]
```

# Check out counts data from folder

```{r}
cMTG<-  read_feather(paste(mtgFolder, "counts.feather",sep=""))
match(names(cMTG),aMTG$sample_id)


# Match by ID
rownames(cMTG) = cMTG$gene
gene = cMTG$gene
cMTG$gene = NULL

# Check if sequence is continuous
identical(seq(1,length(names(cMTG))), match(names(cMTG), aMTG$sample_id))


#aMTG<- aMTG[match(dMTG$sample_id,aMTG$sample_id),]
# Select Subclass
cMTG<-cMTG[,is.element(aMTG$subclass_label,subclass)]

# Looks like this data is already transposed


```



```{r shortcutData}
# Load data from 
dMTG<- read_feather("~/proj/proj5HT/data-raw/L5L23_dMTG.feather")
aMTG<- read_feather("~/proj/proj5HT/data-raw/L5L23_annoMTG.feather")
dPatch<- read_feather("~/proj/proj5HT/data-raw/L5L23_dPatch.feather")
aPatch<- read_feather("~/proj/proj5HT/data-raw/L5L23_aPatch.feather")

```

# Seurat object

### Building the object
data from transcriptomics reads has to have the rownames as the genes, and colnames as the cell identifiers
metadata for input has to have rownames to match the colnames of the transcriptomics data.


```{r}

# Drop NAs from each
dMTG = dMTG[,!is.na(match(colnames(dMTG), colnames(dPatch)))]
dPatch = dPatch[,!is.na(match(colnames(dPatch), colnames(dMTG)))]


# Rearrange dataframe so the column titles match
dPatch = dPatch[,match(colnames(dMTG), colnames(dPatch))]

dMTG = as.data.frame(dMTG)
dPatch = as.data.frame(dPatch)

#### Select shared colnames ####
#dMTG = as.data.frame(dMTG[,colnames(dMTG) %in% colnames(dPatch)])
#dPatch = as.data.frame(dPatch[,colnames(dPatch) %in% colnames(dMTG)])
#dMTG = as.data.frame(dMTG[rownames(dMTG) %in% rownames(dPatch),])

colnames(dPatch)
rownames(dPatch)
str(dPatch)



colnames(dMTG)
rownames(dMTG)
str(dMTG)



#### Set rownames and colnames before transposing ####
rownames(dPatch) = dPatch$sample_id
str(dPatch)
dPatch$sample_id = NULL


#$tst = as.matrix.data.frame(dPatch)
#str(tst)
#tst = as.matrix.data.frame(dPatch)


rownames(dMTG) = dMTG$sample_id
dMTG$sample_id = NULL

# convert dMTG to numeric
#dMTG = apply(dMTG,2,as.numeric)

# Set colnames because the converting to numeric drops them
#colnames(dMTG) = colnames(dPatch)


# t() does not drop row and column names.
dMTG = as.data.frame(t(dMTG))
rownames(tst)
colnames(tst)

str(dMTG)
rownames(dMTG)
colnames(dMTG)

dPatch = as.data.frame(t(dPatch))
rownames(dPatch)
colnames(dPatch)

```

# Construct data object

```{r}
brain.data <- cbind(dMTG, dPatch)  # Include only genes subsetted above

brain.metadata <- data.frame(
  set = c(rep("FACs-Data", dim(dMTG)[2]), rep("PatchSeq", dim(dPatch)[2])),
  subclass = c(
    aMTG$subclass_label,
    aPatch$subclass_Corr_label
  ),
  QC = c(rep("none", dim(dMTG)[2]), aPatch$Norm_Marker_Sum.0.4_label),
  NMS = c(rep(1, dim(dMTG)[2]), aPatch$marker_sum_norm_label),
  #area = c(rep("TCx", dim(dMTG)[2]), aPatch$Cortical_area),
  #projection = c(rep("unknown", dim(dMTG)[2]), aPatch$projection_type),
 # genes_detected = c(rep(0, dim(dMTG)[2]), aPatch$Genes.Detected),
  tree_score = c(rep(0, dim(dMTG)[2]), aPatch$score.Tree_label),
  corr_score = c(rep(0, dim(dMTG)[2]), aPatch$score.Corr_label),
  # intron_reads = c(
  #   rep(0, dim(dMTG)[2]),
  #   aPatch$percent_reads_aligned_to_introns
  # ),
  subclass_corr = c(
    aMTG$subclass_label,
    aPatch$subclass_Corr_label
  )#,
  #prep = c(rep("acute", dim(dMTG)[2]), aPatch$Prep)
)
rownames(brain.metadata) <- colnames(brain.data)

## Construct data set lists
brain      <- CreateSeuratObject(counts = brain.data, meta.data = brain.metadata, project = "5HT")
#brain[["RNA"]]$data <- NormalizeData(brain[["RNA"]]$counts)
#brain.list <- SplitObject(object = brain, split.by = "set")
```

### Do a percentage feature set for an initial observation of the data.  

```{r}
brain[["percent.mt"]] <- PercentageFeatureSet(brain, pattern = "^MT*")
```

#### Figures

Combined shows nCount_RNA being far from each other
```{r, eval = TRUE}
VlnPlot(brain, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```


Separating by method.
```{r, eval = TRUE}
VlnPlot(brain, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, split.by = "set") + theme(legend.position = "bottom")
```


Loading the data.feather from the ref tax, and the data.feather from the patchseq data shows the counts are not properly scaled?

# Construct data object with counts

```{r}
cMTG<- read_feather("data-raw/L5L23_cMTG.feather")
cMTG = as.data.frame(cMTG)
rownames(cMTG) = cMTG$gene
cMTG$gene

if (any(is.na(match(rownames(cMTG), rownames(dPatch))))) {
  cMTG = cMTG[!is.na(match(rownames(cMTG), rownames(dPatch))), ]
}
cMTG$gene = NULL

cMTG = log(cMTG+1)
dPatch = log(dPatch+1)


```


```{r}
brain.data <- cbind(cMTG, dPatch)  # Include only genes subsetted above

brain.metadata <- data.frame(
  set = c(rep("FACs-Counts", dim(cMTG)[2]), rep("PatchSeq", dim(dPatch)[2])),
  subclass = c(
    aMTG$subclass_label,
    aPatch$subclass_Corr_label
  ),
  QC = c(rep("none", dim(cMTG)[2]), aPatch$Norm_Marker_Sum.0.4_label),
  NMS = c(rep(1, dim(cMTG)[2]), aPatch$marker_sum_norm_label),
  #area = c(rep("TCx", dim(dMTG)[2]), aPatch$Cortical_area),
  #projection = c(rep("unknown", dim(dMTG)[2]), aPatch$projection_type),
 # genes_detected = c(rep(0, dim(dMTG)[2]), aPatch$Genes.Detected),
  tree_score = c(rep(0, dim(cMTG)[2]), aPatch$score.Tree_label),
  corr_score = c(rep(0, dim(cMTG)[2]), aPatch$score.Corr_label),
  # intron_reads = c(
  #   rep(0, dim(dMTG)[2]),
  #   aPatch$percent_reads_aligned_to_introns
  # ),
  subclass_corr = c(
    aMTG$subclass_label,
    aPatch$subclass_Corr_label
  )#,
  #prep = c(rep("acute", dim(dMTG)[2]), aPatch$Prep)
)
rownames(brain.metadata) <- colnames(brain.data)

## Construct data set lists
brain      <- CreateSeuratObject(counts = brain.data, meta.data = brain.metadata, project = "5HT")
#brain[["RNA"]]$data <- NormalizeData(brain[["RNA"]]$counts)
#brain.list <- SplitObject(object = brain, split.by = "set")
```

### Do a percentage feature set for an initial observation of the data.  

```{r}
brain[["percent.mt"]] <- PercentageFeatureSet(brain, pattern = "^MT*")
```

#### Figures

Combined shows nCount_RNA being far from each other
```{r, eval = TRUE}
VlnPlot(brain, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```


Separating by method.
```{r, eval = TRUE}

VlnPlot(brain, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, split.by = "set") + theme(legend.position = "bottom")
brain@assays$RNA$counts

```

# 

```{r}

#brain.list <- SplitObject(object = brain, split.by = "set")

#brain.list$PatchSeq[["RNA"]]$data = Seurat::NormalizeData(brain.list$PatchSeq[["RNA"]]$counts, scale.factor = 1e6)
#VlnPlot(brain.list$PatchSeq, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
#VlnPlot(brain.list$`FACs-Counts`, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

```

```{r}

#VlnPlot(brain.list$PatchSeq, features = HTRgenes12, pt.size = 0.2, ncol = 3)

VlnPlot(brain, features = HTRgenes12, pt.size = 0.2, ncol = 3, split.by = "set")

brain[["RNA"]]$data <- NormalizeData(brain[["RNA"]]$counts)
VlnPlot(brain, features = HTRgenes12, pt.size = 0.2, ncol = 3, split.by = "set")

```

